python/cudf/cudf/core/column/lists.py

```diff
@@ -228,3 +230,53 @@ def len(self):
         dtype: int32
         """
         return self._return_or_inplace(count_elements(self._column))
+
+    def take(self, lists_indices):
+        """
+        Collect list elements based on given indices.
+
+        Parameters
+        ----------
+        lists_indices: List type arrays
+            Specifies what to collect from each row
+
+        Returns
+        -------
+        ListColumn
+
+        Examples
+        --------
+        >>> s = cudf.Series([[1, 2, 3], None, [4, 5]])
+        >>> s
+        0    [1, 2, 3]
+        1         None
+        2       [4, 5]
+        dtype: list
+        >>> s.list.take([[0, 1], [], []])
+        0    [1, 2]
+        1      None
+        2        []
+        dtype: list
+        """
+
+        lists_indices_col = as_column(lists_indices)
+        if not isinstance(lists_indices_col, ListColumn):
+            raise ValueError("lists_indices should be list type array.")
+        if not lists_indices_col.size == self._column.size:
+            raise ValueError(
+                "lists_indices and list column is of different " "size."
+            )
+        if lists_indices_col.null_count > 0:
+            raise ValueError("lists_indices contains null elements.")
```

Comment:
Just want to make a note that this check isn't free. Should we be checking this or defer to libcudf behavior here?

Reply:
Yes, I think so - currently it's doing a double check. Should I wrap the cython call in `try` block?

Reply:
How would it raise through libcudf? Do we need to catch the exception that Cython will raise for us?

Reply:
https://github.com/rapidsai/cudf/blob/0b766c5a1de627ad70a6d83e167104a52db1565b/cpp/src/lists/copying/segmented_gather.cu#L37

It checks through `has_nulls` (which calls `null_count`) and raises `cudf::logic_error` with the error strings. I saw similar error handling here for example:
https://github.com/rapidsai/cudf/blob/0b766c5a1de627ad70a6d83e167104a52db1565b/python/cudf/cudf/core/column/column.py#L735-L740

Reply:
Sounds good. Lets do similar with a `try / except` then.